ndb/src/google/cloud/ndb/key.py

```diff
@@ -801,7 +810,10 @@ def _project_from_app(app, allow_empty=False):
     if app is None:
         if allow_empty:
             return None
-        app = os.environ.get(_APP_ID_ENVIRONMENT, _APP_ID_DEFAULT)
+        client = _runstate.current().client
+        app = client.project
+        if app is None:
+            app = os.environ.get(_APP_ID_ENVIRONMENT, _APP_ID_DEFAULT)
```

Comment:
FWIW `_APP_ID_ENVIRONMENT` is irrelevant to non-"GAE standard"

Reply:
Good to know, thanks.